### PR TITLE
feat: add config reloader

### PR DESCRIPTION
**Pull request by huanggze**
Signed-off-by: huanggze <loganhuang@yunify.com>

Merge configmap-reloader and fluentbit-controller functionalities into one, so we don't need a configmap-reloader container.

This cooperates with the new Fluent Bit Operator.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+	if err := g.Run(); err != nil {
+		level.Error(logger).Log("err", err)
+		os.Exit(1)
+	}
+	level.Info(logger).Log("msg", "See you next time!")
+}
+
+func isValidEvent(event fsnotify.Event) bool {
+	if event.Op&fsnotify.Create != fsnotify.Create {
```
Comment (benjaminhuo) [inline main.go:139]: non-create event like modification isn't a valid event while create event is a valid event?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		return false
+	}
+	if filepath.Base(event.Name) != "..data" {
```
Comment (benjaminhuo) [inline main.go:142]: base path ..data means ?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+							if !isValidEvent(event) {
+								continue
+							}
+
+							level.Info(logger).Log("msg", "Config file changed")
+							level.Info(logger).Log("msg", "Stop Fluent Bit")
+
+							if err := cmd.Process.Kill(); err != nil {
+								level.Error(logger).Log("err", err)
+							}
```
Comment (benjaminhuo) [inline main.go:92]: Is it better to put repeated use of lines of cmd related statements into functions like startFluentBit, restartFluentBit ?

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
@huanggze

File: main.go (hunk 1 above)
Comment (huanggze) [reply main.go:139]: The code here is heavily inspired by jimmidyson/configmap-reload. I put a comment above this method to claim that we are using their code.

https://github.com/jimmidyson/configmap-reload/blob/master/configmap-reload.go#L162-L170

File: main.go (hunk 3 above)
Comment (huanggze) [reply main.go:92]: It think what we have is good, because it is a small code snippet with few lines

Comment (benjaminhuo) [reply main.go:92]: ok

File: main.go (hunk 1 above)
Comment (benjaminhuo) [reply main.go:139]: ok，we have to do through test on this

**Pull request merged**